novawallet/Common/Services/Web3AlertService/PushNotificationsStatusService.swift

```diff
@@ -107,9 +107,16 @@ final class PushNotificationsStatusService: NSObject {
         }
     }
 
+    private func syncSettingsIfNeeded(newStatus: PushNotificationsStatus) {
+        if newStatus == .denied, settingsManager.notificationsEnabled {
+            settingsManager.notificationsEnabled = false
+        }
+    }
+
     private func updateStatus() {
         status { [weak self] newStatus in
             self?.statusObservable.state = newStatus
+            self?.syncSettingsIfNeeded(newStatus: newStatus)
```

Comment:
I would first sync the settings and then update the status because status update triggers observer which in turn can access the settings